README.md

```diff
@@ -30,6 +30,24 @@ $ shortlinks --listen :8081 --db file:shortlinks.db
 
 Then navigate to `http://localhost:8081` and create your first shortlink!
 
+## Variables
+
+In addition to simple links, variables can be added to a link to be filled out based on what the input URL is.
+
+Variables are added with `%s` in the URL. 
+For instance if your link is defined as `https://atlassian.net/browse/%s` and `j` pointed to it,
+then `s.frew.co/j` would resolve to `https://atlassian.net/browse/` and `s.few.co/j/JIRA-000` would resolve to `https://atlassian.net/browse/JIRA-000`.
+
+This works with multiple variables in a left to right fashion.
+If you define `https://some.url/q=%s&q=%s` with `some` pointed to it,
+then `s.frew.co/some/firstquery/secondquery` would resolve to `https://some.url/q=firstquery&q=secondquery`.
+
+Siumilarly, extra variables are appended to the end of the url with slashes seperating them.
+If you define `https://other.url/%s/something` with `other` pointed to it,
+then `s.frew.co/other/this/is/appended/to/the/end` will resolve to `https://other.url/this/something/is/appended/to/the/end`
+
+More examples of this are in the `handler_index_test.go` file.
```

Comment:
Maybe timebox an effort to have the tests driven from examples in the doc, rather than mentioning a source file in the doc, if that makes sense? The doc could be both here and in the UI or something, but consider that this is not solely used by engineers but people who write literally zero code

Reply:
I'm not sure I follow what you mean by tests driven from examples in the doc since all these examples are already included in the table test.

Reply:
I mean what if you had a markdown table here, and the tests read the README.md to create the table of tests, so that people can see the intentions.  Just a representative sample.

Reply:
I updated the readme to include a table, but the test has addtitional cases so having the test parse the md file doesnt seem to make sense here.